src/Elastic.Markdown/Myst/InlineParsers/DiagnosticLinkInlineParser.cs

```diff
@@ -100,6 +101,27 @@ private static void ValidateAndProcessLink(LinkInline link, InlineProcessor proc
 	{
 		var url = link.Url;
 
+		if (url?.Contains("{{") == true)
+		{
+			var replacedUrl = url.ReplaceSubstitutions(processor.GetContext());
+			if (replacedUrl.Contains("{{"))
+			{
+				processor.EmitWarning(link,
+					"The url contains a template expression. Please do not use template expressions in links. " +
```

Comment:
I think the error message here is misleading, we allow template expressions but this indicates a substitution key was not found and thus left the expression in the url.

Reply:
https://github.com/elastic/docs-builder/pull/715/commits/de5933b8cbe88fcbe0d1b6e136536297d00a185c